### PR TITLE
Merge auth-exp into master

### PR DESCRIPTION
**Pull request by samhorlbeck**

### CODE REVIEW

File: packages-exp/auth-compat-exp/demo/public/common.js
Comment (samhorlbeck) [inline packages-exp/auth-compat-exp/demo/public/common.js:1]: This is copied verbatim from `packages/auth/demo/public/common.js`

File: packages-exp/auth-compat-exp/demo/public/script.js
Comment (samhorlbeck) [inline packages-exp/auth-compat-exp/demo/public/script.js:1]: This is copied verbatim from `packages/auth/demo/public/script.js`

**Comment by samhorlbeck** [comment]
Note: all of this code was already peer-reviewed by folks on the immediate team

**Comment by schmidt-sebastian** [comment]
@samhorlbeck Are there particular areas of the code that you would like us (me) to look at? That would help me up my enthusiasm to look at this :)

**Comment by samhorlbeck** [comment]
@schmidt-sebastian I think the places where we were most out of our depth: so things like the build system/setup (Fei just merged in a fix cleaning this up as well). Perhaps also the way we're bubbling up our re-exports/different platform targeting.

Also maybe check over the code that actually makes the network calls (auth-exp/src/core/api/index.ts). Here's a deep link. It doesn't work for me since the file was auto-collapsed but maybe it'll work for you. https://github.com/firebase/firebase-js-sdk/pull/3722/files#diff-16ae36e512ead2eb1e0c14f0f1bf7dd3R18

@avolkovi anything else you want Sebastian to look over specifically

You can ignore anything in the demo/ folders I think.

**Comment by avolkovi** [comment]
@schmidt-sebastian would also love your eyes on some of the persistence magic, especially how we're chatting with indexedDB

**Comment by schmidt-sebastian** [comment]
Very crude review: https://gist.github.com/schmidt-sebastian/7288e04f078fc057629b6a2395150001

**Comment by samhorlbeck** [comment]
@schmidt-sebastian I just opened https://github.com/firebase/firebase-js-sdk/pull/3771 to address the network timeout bit (figured it'd be easier to review in a separate PR)

**Comment by Feiyang1** [comment]
Is there anything blocking the merge that I can help with?

**Comment by samhorlbeck** [comment]
Yeah @Feiyang1 would you mind double checking our `package.json` files? I also have https://github.com/firebase/firebase-js-sdk/pull/3783 that fixes `yarn test` in auth-compat-exp

**Comment by Feiyang1** [comment]
@samhorlbeck Didn't see any issue except needing some dep version updates. I opened a PR https://github.com/firebase/firebase-js-sdk/pull/3809. Let me know right before the merge, so I can update the versions one more time and merge my PR to this one.

**Review by Feiyang1 — APPROVED** [review]

**Review by avolkovi — APPROVED** [review]

**Pull request merged**
